### ISSUE TITLE
proposal: x/net/http3: add HTTP/3 implementation

### ISSUE THREAD

**Issue by neild** [golang/go#70914]
This is a proposal to add an HTTP/3 client and server implementation in x/net/http3.

Similarly to the accepted form of #58547 (x/net/quic), this proposal is to add an experimental package with an API subject to change during development. Once we have a complete implementation, we will file a separate proposal for API review.

Initial development will be in an internal package (x/net/internal/http3) until the details are firm enough for external testing. We will then move the package to x/net/http3 and (when we have some confidence the API is right) file the API review proposal.

**Comment by aclements** [comment]
This proposal has been added to the [active column](https://go.dev/s/proposal-status#active) of the proposals project
and will now be reviewed at the weekly proposal review meetings.
— aclements for the proposal review group

**Comment by aclements** [comment]
Just to clarify, is the intent to put x/net/http3 behind a GOEXPERIMENT and have a separate proposal for stabilizing the API?

**Comment by aclements** [comment]
It turns out it's annoying to put this behind a GOEXPERIMENT because the list of experiments is managed in the main repo.

We could use a build tag instead.

Another option is x/exp, but that's much more painful for users to migrate away from once this is stabilized, and we'd lose all the commit history once it moved to x/net.

**Comment by asoseil** [comment]
Putting it under a build tag makes sense. Could we target this for the next minor release?

**Comment by aclements** [comment]
I assume you mean the next minor release of the x/net module. The proposal needs to be accepted first (I don't foresee any obstacles to that, but there's some delay built into the process to allow for comments), and then I suspect it will land in the next minor release of x/net following that.

**Comment by aclements** [comment]
**[Placed on hold](https://go.dev/s/proposal-status#hold)**.
— aclements for the proposal review group

**Comment by philoserf** [comment]
> ... with a note of what it is waiting on.

**Comment by FZambia** [comment]
Could someone please clarify the reason of moving to HOLD status? Not very obvious 🙏

**Comment by neild** [comment]
On hold pending a concrete proposal for the exported package API.
